### PR TITLE
[#ICC-293 ] Use template for email sent to notify a message

### PR DESCRIPTION
**Pull request by fabriziopapi**
#### List of Changes

see commit history

#### Motivation and Context

The email used to notify a new message is created composing magic string hard-coded in this repo. With this PR, we added a new generation script that import the email content template from another repository: this imported template will be used to generate the email content.
The use of the new template-based email is behind a feature flag.

#### How Has This Been Tested?

unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: EmailNotification/__tests__/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    const mockProcessor = jest.fn().mockImplementationOnce(async () => {
+      throw anError;
+    });
+    const result = await contentToHtml(mockProcessor)("")();
+    expect(result).toEqual(E.left(anError));
+  });
+});
+
+describe("messageToHtml", () => {
+  test("GIVEN a simple message WHEN is converted to an html using the default processor THEN return an either contianing the email template with the simple content", async () => {
```
Comment (AleDore) [inline EmailNotification/__tests__/utils.test.ts:35]:
```suggestion
  test("GIVEN a simple message WHEN is converted to an html using the default processor THEN return an either containing the email template with the simple content", async () => {
```

File: EmailNotification/__tests__/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const simpleMessage = {
+      content: aMessageContent,
+      senderMetadata: aCreatedMessageEventSenderMetadata
+    };
+    const result = await messageToHtml()(simpleMessage)();
+    expect(result).toMatchSnapshot();
+  });
+
+  test("GIVEN a sender with a leading zero WHEN is converted to an html using the default processor THEN return an either contianing the email template with the simple content", async () => {
```
Comment (AleDore) [inline EmailNotification/__tests__/utils.test.ts:44]: ditto

File: EmailNotification/handler.ts [hunk 3]
```diff
@@ -126,12 +139,30 @@ export const getEmailNotificationHandler = (
...
+                  throw err;
+                }),
+                TE.toUnion
+              )()
+            : generateDocumentHtml(
+                content.subject,
+                content.markdown,
+                senderMetadata
+              ));
+
```
Comment (AleDore) [inline EmailNotification/handler.ts:165]: why not collapsing all in the same pipe?

**Comment by gquadrati** [comment]
Code review pipeline is failing (Build step)

File: EmailNotification/__tests__/utils.test.ts (hunk 1 above)
Comment (fabriziopapi) [reply EmailNotification/__tests__/utils.test.ts]: fixed, ty

File: EmailNotification/handler.ts (hunk 3 above)
Comment (fabriziopapi) [reply EmailNotification/handler.ts]: fixed, ty

**Comment by fabriziopapi** [comment]
> Code review pipeline is failing (Build step)

seems that not all the azure pipeline run on the same environment configuration. 
Fixed to new generation script to support multiple environment, ty.

**Review by AleDore — APPROVED** [review]

**Pull request merged**